From 02b406e82dfde895c03a03e9016844138a2ea421 Mon Sep 17 00:00:00 2001
From: Ken Zhu <quic_guigenz@quicinc.com>
Date: Thu, 14 Oct 2021 08:50:03 -0700
Subject: [PATCH 232/500] net: fix potential memory leak

When A packets come from a disabled port, if it is not a accepted
EAPOL, it has no chance to get freed.

Change-Id: Ib2bddeb53f2117bcc803d11912471a8456a3148a
Signed-off-by: Ken Zhu <quic_guigenz@quicinc.com>
---
 net/bridge/br_input.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/bridge/br_input.c b/net/bridge/br_input.c
index fcdd91002d8a..93c2ed33a1c9 100644
--- a/net/bridge/br_input.c
+++ b/net/bridge/br_input.c
@@ -452,7 +452,7 @@ static rx_handler_result_t br_handle_frame(struct sk_buff **pskb)
 				return RX_HANDLER_PASS;
 			}
 		}
-		break;
+		goto drop;
 
 	case BR_STATE_FORWARDING:
 	case BR_STATE_LEARNING:
-- 
2.34.1

